packages/api/endpoints/granules.js

```diff
@@ -44,6 +44,7 @@ async function list(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function put(req, res) {
+  const cmrRevisionId = req.headers['cmr-revision-id'];
```

Comment:
I understand the changes to `post-to-cmr` below in the context of the ticket, but I'm not sure I understand why we're updating the granules endpoint to take a CMR revision ID, or the changes to rules. We would need these changes If PODAAC wanted to specify a CMR revision ID as input to a workflow for a granule, but my understanding of the ticket was that the tasks **within** their workflow will pull the CMR revision ID for a granule and send it to post-to-cmr.

Reply:
Hm I am sure I am misunderstanding how to achieve this. I was under the assumption that it would work in the following way:
```
PUT granule -> applyWorkflow -> Rule.buildPayload (takes in cmrRevisionId) -> PostToCmr
```

Is what I am missing is the `post-to-cmr` will be invoked in another way?

Reply:
The CMR is usually generated by the end users within their processing tasks. The processing tasks are completely defined by them. In our fake workflow we generate our own. 

I think the use case here is that as they mess with their own CMR files they will tracking the revision-id within their workflow tasks and thus just needs to be an input to the post to cmr task.